### DISCUSSION TITLE
(Mobile) How to detect when the user switches chain in metamask

### DISCUSSION THREAD

**Discussion by ezra-sg** [WalletConnect/walletconnect-monorepo#3225] [Developer Support]
Hello!

I am having trouble detecting when the user switches network in metamask. I am currently using wagmi to attempt to do so, like this: 
```ts
watchNetwork((network) => {
    console.log(network.chain);
});
```
_Occasionally_, this callback will fire as expected, logging the new chain. however this works very infrequently.

additionally, using `getNetwork`, the chain returned is stale (which is to say, it does not reflect changes in the metamask app's selected network)
```ts
import { getNetwork } from '@wagmi/core'
 
const { chain, chains } = getNetwork()
```

am I missing something, or thinking about this incorrectly? the use case is that I need to check the `chainID` before executing an action, such as a send transaction, so the user (of course) doesn't send ETH on the wrong network. 

I notice that, when the user is initially on the wrong network, and I run `prepareSendTransaction`, _sometimes_ the metamask app will open and ask the user to switch chains. I am wondering if there is some way to programmatically trigger this behavior after correctly detecting what network the user has selected in metamask.

attempting to switch the network like so does not appear to do anything at all:
```ts
import { switchNetwork } from '@wagmi/core'
 
const network = await switchNetwork({
  chainId: 40,
})
```

thank you so much for taking the time to read this :)

**Comment by glitch-txs** [comment]
Seems to be an issue on MetaMask side https://github.com/MetaMask/metamask-mobile/issues/6655

**Reply by ezra-sg** [reply]
Damn okay well thank you
